openqaoa/problems/problem.py

```diff
@@ -763,49 +736,18 @@ def get_qubo_problem(self):
         n_variables = self.n_items + n_variables_slack
         terms, weights = self.terms_and_weights()
 
-        #         ising_terms, ising_coeffs = terms,weights
-        ising_terms, ising_coeffs = [], []
-
-        constant_term = 0
-        linear_terms = np.zeros(n_variables)
-
-        # Process the given terms and weights
-        for weight, term in zip(weights, terms):
-
-            if len(term) == 2:
-                u, v = term
+        # ising_terms, ising_weights = terms,weights
+        ising_terms, ising_weights = convert_binary_to_ising(terms, weights)
 
-                if u != v:
-                    ising_terms.append([u, v])
-                    ising_coeffs.append(weight / 4)
-                else:
-                    constant_term += weight / 4
-
-                linear_terms[term[0]] -= weight / 4
-                linear_terms[term[1]] -= weight / 4
-                constant_term += weight / 4
-            elif len(term) == 1:
-                linear_terms[term[0]] -= weight / 2
-                constant_term += weight / 2
-            else:
-                constant_term += weight
-
-        for variable, linear_term in enumerate(linear_terms):
-            ising_terms.append([variable])
-            ising_coeffs.append(linear_term)
-
-        ising_terms.append([])
-        ising_coeffs.append(constant_term)
-
-        return QUBO(n_variables, ising_terms, ising_coeffs)
+        return QUBO(n_variables, ising_terms, weights)
```

Comment:
is this supposed to be `QUBO(n_variables, ising_terms, ising_weights)` ?

Reply:
Yes !